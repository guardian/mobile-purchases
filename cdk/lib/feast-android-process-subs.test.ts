import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {FeastAndroidProcessSubscriptions } from "./feast-android-process-subs";

describe('The FeastAndroidSubscriptions stack', () => {
    it('matches the snapshot', () => {
        const app = new App();
        const codeStack = new FeastAndroidProcessSubscriptions(app, 'feast-subscriptions-CODE',{ stack: 'mobile', stage: 'CODE' });
        const prodStack = new FeastAndroidProcessSubscriptions(app, 'feast-subscriptions-PROD',{ stack: 'mobile', stage: 'PROD' });

        expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
        expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
    });
});