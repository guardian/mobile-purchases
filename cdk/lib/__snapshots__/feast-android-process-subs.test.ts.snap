// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FeastAndroidSubscriptions stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuSnsLambdaExperimental",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "feastandroidprocesssubsCODEprocessinglambdaTopicName357693EC": {
      "Value": {
        "Fn::GetAtt": [
          "feastandroidprocesssubsCODEsnsqueueD30D1499",
          "QueueName",
        ],
      },
    },
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "feastandroidprocesssubsCODEprocessinglambdaA59E895B": {
      "DependsOn": [
        "feastandroidprocesssubsCODEprocessinglambdaServiceRoleDefaultPolicy804F6A7D",
        "feastandroidprocesssubsCODEprocessinglambdaServiceRoleE69B0532",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "mobile/CODE/feast-android-process-subs-CODE-lambda/feast-android-process-subs.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "feast-android-process-subs-CODE-lambda",
            "STACK": "mobile",
            "STAGE": "CODE",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsCODEprocessinglambdaServiceRoleE69B0532",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-android-process-subs-CODE-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "feastandroidprocesssubsCODEprocessinglambdaAllowInvokefeastsubscriptionsCODEfeastandroidprocesssubsCODEprocessinglambdaSnsExistingIncomingEventsTopic7FA37995F906FF12": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsCODEprocessinglambdaA59E895B",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastandroidprocesssubsCODEsnsqueueD30D1499",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "feastandroidprocesssubsCODEprocessinglambdaServiceRoleDefaultPolicy804F6A7D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/mobile/CODE/feast-android-process-subs-CODE-lambda/feast-android-process-subs.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/mobile/feast-android-process-subs-CODE-lambda",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/mobile/feast-android-process-subs-CODE-lambda/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "feastandroidprocesssubsCODEprocessinglambdaServiceRoleDefaultPolicy804F6A7D",
        "Roles": [
          {
            "Ref": "feastandroidprocesssubsCODEprocessinglambdaServiceRoleE69B0532",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "feastandroidprocesssubsCODEprocessinglambdaServiceRoleE69B0532": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-android-process-subs-CODE-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "feastandroidprocesssubsCODEprocessinglambdafeastandroidprocesssubsCODEprocessinglambdaSnsExistingIncomingEventsTopicC7C73133": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsCODEprocessinglambdaA59E895B",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "Region": {
          "Ref": "AWS::Region",
        },
        "TopicArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastandroidprocesssubsCODEsnsqueueD30D1499",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "feastandroidprocesssubsCODEsnsqueueD30D1499": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "feast-android-process-subs-CODE-queue",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;

exports[`The FeastAndroidSubscriptions stack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuSnsLambdaExperimental",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "feastandroidprocesssubsPRODprocessinglambdaTopicName91452A43": {
      "Value": {
        "Fn::GetAtt": [
          "feastandroidprocesssubsPRODsnsqueue9DF14337",
          "QueueName",
        ],
      },
    },
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "feastandroidprocesssubsPRODprocessinglambdaAllowInvokefeastsubscriptionsPRODfeastandroidprocesssubsPRODprocessinglambdaSnsExistingIncomingEventsTopic7AF3D801B2933E40": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsPRODprocessinglambdaD6198782",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastandroidprocesssubsPRODsnsqueue9DF14337",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "feastandroidprocesssubsPRODprocessinglambdaD6198782": {
      "DependsOn": [
        "feastandroidprocesssubsPRODprocessinglambdaServiceRoleDefaultPolicy819E0CA5",
        "feastandroidprocesssubsPRODprocessinglambdaServiceRoleA71931F6",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "mobile/PROD/feast-android-process-subs-PROD-lambda/feast-android-process-subs.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "feast-android-process-subs-PROD-lambda",
            "STACK": "mobile",
            "STAGE": "PROD",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsPRODprocessinglambdaServiceRoleA71931F6",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-android-process-subs-PROD-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "feastandroidprocesssubsPRODprocessinglambdaServiceRoleA71931F6": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-android-process-subs-PROD-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "feastandroidprocesssubsPRODprocessinglambdaServiceRoleDefaultPolicy819E0CA5": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/mobile/PROD/feast-android-process-subs-PROD-lambda/feast-android-process-subs.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/mobile/feast-android-process-subs-PROD-lambda",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/mobile/feast-android-process-subs-PROD-lambda/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "feastandroidprocesssubsPRODprocessinglambdaServiceRoleDefaultPolicy819E0CA5",
        "Roles": [
          {
            "Ref": "feastandroidprocesssubsPRODprocessinglambdaServiceRoleA71931F6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "feastandroidprocesssubsPRODprocessinglambdafeastandroidprocesssubsPRODprocessinglambdaSnsExistingIncomingEventsTopic11B40574": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "feastandroidprocesssubsPRODprocessinglambdaD6198782",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "Region": {
          "Ref": "AWS::Region",
        },
        "TopicArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastandroidprocesssubsPRODsnsqueue9DF14337",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "feastandroidprocesssubsPRODsnsqueue9DF14337": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "feast-android-process-subs-PROD-queue",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
